ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/ExplicitDisclosureIT.scala

```diff
@@ -48,134 +47,164 @@ final class ExplicitDisclosureIT extends LedgerTestSuite {
-      dummyCreate = createdEvents(dummyTxs(0)).head
-      dummyDisclosedContract = createEventToDisclosedContract(dummyCreate)
+        dummyCid <- ownerParticipant.create(owner, Dummy(owner))
+        dummyTxs <- ownerParticipant.flatTransactionsByTemplateId(Dummy.id, owner)
+        dummyCreate = createdEvents(dummyTxs(0)).head
+        dummyDisclosedContract = createEventToDisclosedContract(dummyCreate)
 
-      // Exercise works with provided disclosed contract
-      _ <- testContext.exerciseFetchDelegated(testContext.disclosedContract)
-      // Exercise works with the Dummy contract as a superfluous disclosed contract
-      _ <- testContext.exerciseFetchDelegated(testContext.disclosedContract, dummyDisclosedContract)
+        // Exercise works with provided disclosed contract
+        _ <- testContext.exerciseFetchDelegated(testContext.disclosedContract)
+        // Exercise works with the Dummy contract as a superfluous disclosed contract
+        _ <- testContext.exerciseFetchDelegated(
+          testContext.disclosedContract,
+          dummyDisclosedContract,
+        )
 
-      // Archive the Dummy contract
-      _ <- ledger.exercise(owner, dummyCid.exerciseArchive())
+        // Archive the Dummy contract
+        _ <- ownerParticipant.exercise(owner, dummyCid.exerciseArchive())
 
-      // Exercise works with the archived superfluous disclosed contract
-      _ <- testContext.exerciseFetchDelegated(testContext.disclosedContract, dummyDisclosedContract)
-    } yield ()
+        // Exercise works with the archived superfluous disclosed contract
+        _ <- testContext.exerciseFetchDelegated(
+          testContext.disclosedContract,
+          dummyDisclosedContract,
+        )
+      } yield ()
   })
 
   test(
     "EDExerciseByKeyDisclosedContract",
     "A disclosed contract can be exercised by key with non-witness readers if authorized",
-    partyAllocation = allocate(TwoParties),
+    partyAllocation = allocate(SingleParty, SingleParty),
     enabled = _.explicitDisclosure,
   ) { implicit ec =>
-    { case Participants(Participant(ledger, owner, divulgee)) =>
-      for {
-        // Create contract with `owner` as only stakeholder
-        _ <- ledger.create(owner, WithKey(owner))
-        withKeyTxIds <- ledger.flatTransactionsByTemplateId(WithKey.id, owner)
-        withKeyCreate = createdEvents(withKeyTxIds(0)).head
-        withKeyDisclosedContract = createEventToDisclosedContract(withKeyCreate)
-        exerciseByKeyError <- ledger
-          .submitAndWait(
-            exerciseWithKey_byKey_request(ledger, owner, divulgee, None)
+    {
+      case Participants(
+            Participant(ownerParticipant, owner),
+            Participant(divulgeeParticipant, divulgee),
+          ) =>
+        for {
+          // Create contract with `owner` as only stakeholder
+          _ <- ownerParticipant.create(owner, WithKey(owner))
+          withKeyTxIds <- ownerParticipant.flatTransactionsByTemplateId(WithKey.id, owner)
```

Comment:
Maybe instead of using flatTx stream it's better to introduce a `otherParticipant.createAndGetTxId` helper that obtains the tx id from the submitAndWait response message?

Reply:
Yeah, these two lines wouldn't haven't warranted introducing a new helper function but good thing that we already have something like this (submitAndWaitForTransaction). Changed, thanks!